app/javascript/controllers/range_limit_controller.js

```diff
@@ -0,0 +1,26 @@
+import { Controller } from "@hotwired/stimulus"
+
+export default class extends Controller {
+  connect() {
+    this.beginInput = this.element.querySelector('input[name="range[date_range][begin]"]');
+    this.endInput = this.element.querySelector('input[name="range[date_range][end]"]');
+    this.beginInput.addEventListener('input', event => this.updateInputRequiredState(event));
+    this.endInput.addEventListener('input', event => this.updateInputRequiredState(event));
+  }
+
+  updateInputRequiredState(event) {
```

Comment:
I want to maintain the user's current ability to reset the date range search by blanking out both sides of the range. You're correct in that my logic could be consolidated though, thanks.

Reply:
I hadn't considered that. Can you add a comment about that use case here?

Reply:
Good idea. I added a comment and a note to check the template because it's only overridden to set the data target. I think it's likely this code gets removed as a result of the production WC.